vault/resource_aws_auth_backend_roletag_blacklist.go

```diff
@@ -0,0 +1,153 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+var (
+	awsAuthBackendRoleTagBlacklistBackendFromPathRegex = regexp.MustCompile("^auth/(.+)/config/tidy/roletag-blacklist$")
+)
+
+func awsAuthBackendRoleTagBlacklistResource() *schema.Resource {
+	return &schema.Resource{
+		Create: awsAuthBackendRoleTagBlacklistWrite,
+		Read:   awsAuthBackendRoleTagBlacklistRead,
+		Update: awsAuthBackendRoleTagBlacklistWrite,
+		Delete: awsAuthBackendRoleTagBlacklistDelete,
+		Exists: awsAuthBackendRoleTagBlacklistExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"backend": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Unique name of the auth backend to configure.",
+				ForceNew:    true,
+				Default:     "aws",
+				// standardise on no beginning or trailing slashes
+				StateFunc: func(v interface{}) string {
+					return strings.Trim(v.(string), "/")
+				},
+			},
+			"safety_buffer": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "The amount of extra time that must have passed beyond the roletag expiration, before it's removed from backend storage.",
+			},
+			"disable_periodic_tidy": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "If true, disables the periodic tidying of the roletag blacklist entries.",
+			},
+		},
+	}
+}
+
+func awsAuthBackendRoleTagBlacklistWrite(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	backend := d.Get("backend").(string)
+	data := map[string]interface{}{}
+
+	if v, ok := d.GetOk("safety_buffer"); ok {
+		data["safety_buffer"] = v
+	}
+	if v, ok := d.GetOk("disable_periodic_tidy"); ok {
+		data["disable_periodic_tidy"] = v
+	}
+
+	path := awsAuthBackendRoleTagBlacklistPath(backend)
+
+	log.Printf("[DEBUG] Configuring AWS auth backend roletag blacklist %q", path)
+	_, err := client.Logical().Write(path, data)
+
+	d.SetId(path)
+
+	if err != nil {
+		d.SetId("")
+		return fmt.Errorf("Error configuring AWS auth backend roletag blacklist %q: %s", path, err)
+	}
+	log.Printf("[DEBUG] Configured AWS backend roletag blacklist %q", path)
+
+	return awsAuthBackendRoleTagBlacklistRead(d, meta)
+}
+
+func awsAuthBackendRoleTagBlacklistRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	path := d.Id()
+
+	backend, err := awsAuthBackendRoleTagBlacklistBackendFromPath(path)
+	if err != nil {
+		return fmt.Errorf("Invalid path %q for AWS auth backend roletag blacklist: %s", path, err)
```

Comment:
if the error returned from `awsAuthBackendRoleTagBlacklistBackendFromPath` is "no path found", should we remove this item from state?

Reply:
I can see an argument either way on this one, and it really depends on how we think the ID is more likely to get into an invalid state--from a coding/server error, or from user error during import. If the former, I think having the value remain in state gives us options for workarounds (even though it means manual edits), but the more I think about it, import support means better workarounds are available if we just drop it from state.

TL;DR: after further thought, I agree.